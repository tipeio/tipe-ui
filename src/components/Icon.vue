<template>
  <div
    :class="size"
    :data-tipe-ui="$options.name"
  />
</template>

<script>
import vueTypes from 'vue-types'

const sizes = ['small', 'medium', 'large']

export default {
  name: 'TipeIcon',
  props: {
    size: vueTypes.oneOf(sizes).def('medium')
  }
}
</script>

<style scoped>
[data-tipe-ui='TipeIcon'] {
  background-color: red;
}

[data-tipe-ui='TipeIcon'].small {
  height: 1rem;
  width: 1rem;
}

[data-tipe-ui='TipeIcon'].medium {
  height: 2rem;
  width: 2rem;
}

[data-tipe-ui='TipeIcon'].large {
  height: 3rem;
  width: 3rem;
}
</style>
