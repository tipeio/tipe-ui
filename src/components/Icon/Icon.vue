<template>
  <object
    :data="src"
    type="image/svg+xml">
    <img :src="src" >
  </object>
</template>

<script>
import vueTypes from 'vue-types'

export default {
  name: 'TipeIcon',
  props: {
    icon: vueTypes.string
  },
  computed: {
    src() {
      return require(`../../icons/${this.icon}.svg`)
    }
  }
}
</script>
