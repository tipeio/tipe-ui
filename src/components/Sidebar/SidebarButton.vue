<template>
  <button
    :data-tipe-ui="$options.name"
    @click="$emit('click', $event)"
  >
    <div class="icon-container">
      <tipe-icon
        :icon="icon"
        height="1.5rem"
        width="1.5rem"
      />
    </div>
  </button>
</template>

<script>
import vueTypes from 'vue-types'
import TipeIcon from '../Icon'

export default {
  name: 'TipeSidebarButton',
  components: {
    TipeIcon
  },
  props: {
    icon: vueTypes.string
  }
}
</script>

<style lang="postcss" scoped>
[data-tipe-ui='TipeSidebarButton'] {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  height: 5rem;
  background-color: var(--purple);
  text-decoration: none;
  outline: none;
  border: none;
  cursor: pointer;
}

.icon-container {
  flex: 0 0 4.6875rem;
  display: flex;
  height: 5rem;
  justify-content: center;
  align-items: center;
}
</style>
