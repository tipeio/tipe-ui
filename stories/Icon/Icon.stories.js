import { storiesOf } from '@storybook/vue'

import TipeIcon from '../../src/components/Icon'

const style = () => ({
  height: '1rem',
  width: '1rem'
})

const icons = [
  'check',
  'document',
  'folder',
  'key',
  'number',
  'share',
  'switch',
  'user',
  'color',
  'edit',
  'gear',
  'link',
  'pin',
  'sign-out',
  'text',
  'view',
  'copy',
  'flow',
  'grid',
  'list',
  'search',
  'star-active',
  'trash',
  'wrench',
  'credit-card',
  'folder-open',
  'group',
  'markdown',
  'media',
  'send',
  'star',
  'user-circle',
  'options',
  'plus',
  'cross',
  'arrow-down',
  'arrow-up',
  'arrow-right',
  'arrow-left'
]

storiesOf('Icon', module).add('icons', () => ({
  components: { TipeIcon },
  computed: { style },
  template:
    '<div>' +
    icons.reduce((template, icon) => {
      return (
        template +
        `<tipe-icon :style="style" icon="${icon}" color="rgb(255,0,0)" />`
      )
    }, '') +
    '</div>'
}))
