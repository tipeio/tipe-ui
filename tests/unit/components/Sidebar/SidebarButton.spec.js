import { shallowMount } from '@vue/test-utils'
import SidebarButton from '@/components/Sidebar/SidebarButton.vue'
import Icon from '@/components/Icon'
import faker from 'faker'

describe('Sidebar', () => {
  describe('<template>', () => {
    it('matches previous snapshot', () => {
      const wrapper = shallowMount(SidebarButton)

      expect(wrapper).toMatchSnapshot()
    })

    it('<root> - has correct data-tipe-iu attibute', () => {
      const wrapper = shallowMount(SidebarButton)

      expect(wrapper.attributes()['data-tipe-ui']).toBe('TipeSidebarButton')
    })

    it('<root> - should be button tag', () => {
      const wrapper = shallowMount(SidebarButton)

      expect(wrapper.is('button')).toBe(true)
    })

    it('<tipe-icon> - should exist', () => {
      const wrapper = shallowMount(SidebarButton)

      expect(wrapper.find(Icon).exists()).toBe(true)
    })
  })

  describe(':props', () => {
    it(':icon - should be passed to Icon component', () => {
      const propsData = { icon: faker.lorem.word() }
      const wrapper = shallowMount(SidebarButton, { propsData })

      expect(wrapper.find(Icon).props().icon).toBe(propsData.icon)
    })
  })

  describe('@events', () => {
    it('@click - should emit click event', () => {
      const wrapper = shallowMount(SidebarButton)

      wrapper.trigger('click')
      expect(wrapper.emitted().click).toBeTruthy()
    })
  })
})
