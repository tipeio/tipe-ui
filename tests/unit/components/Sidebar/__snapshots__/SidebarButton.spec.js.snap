// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar <template> matches previous snapshot 1`] = `Promise {}`;
